code-challenges.test.js

```diff
@@ -14,32 +14,102 @@
 
 // a) Create a test with expect statements using the variables provided.
 
+describe("secretMsg", () => {
 const secretCodeWord1 = "Lackadaisical"
 // Expected output: "L4ck4d41s1c4l"
 const secretCodeWord2 = "Gobbledygook"
 // Expected output: "G0bbl3dyg00k"
 const secretCodeWord3 = "Eccentric"
 // Expected output: "3cc3ntr1c"
+    it("takes in a string and returns a string with a coded message", () => {
+        expect(secretMsg(secretCodeWord1)).toEqual("L4ck4d41s1c4l")
+        expect(secretMsg(secretCodeWord2)).toEqual("G0bbl3dyg00k")
+        expect(secretMsg(secretCodeWord3)).toEqual("3cc3ntr1c")
+    })
+})
+
+// output: ReferenceError: secretMsg is not defined
 
 // b) Create the function that makes the test pass.
+    // pseudo code:
+        // use for loop to iterate
+        // use conditionals to match each letter or uppercase letter
+        // .replace method to replace index of string with respective number
+
+    // const secretMsg = (string) => {
+    // for(let i = 0; i < string.length; i++) {
+    //     if (string[i] === 'a' || string[i] === 'A') {
+    //     string = string.replace(string[i], '4')
+    //     } else if (string[i] === 'e' || string[i] === 'E') {
+    //     string = string.replace(string[i], '3')
+    //     } else if (string[i] === 'i' || string[i] === 'I') {
+    //     string = string.replace(string[i], '1')
+    //     } else if (string[i] === 'o' || string[i] === 'O') {
+    //     string = string.replace(string[i], '0')
+    //     }
+    // }
+    // return string;
+    // }
+
+    const secretMsg = (string) => {
+        return string.replace(/[aeio]/gi, (match) => {
```

Comment:
Nice refactor and realizing that you don't need conditionals if you're going to use .replace!